### PR TITLE
RU translation: /tutorials/keras/

### PR DESCRIPTION
**Pull request by 0101011**
1. For convenience I've created a separate **site/ru** folder (site/ru/tutorials/keras/) for all translations to be committed there

2. Feel free to change the structure as needed

### CODE REVIEW

**Comment by 0101011** [comment]
Signed the CLA

**Comment by lamberta** [comment]
Thanks, Andrew! That was quicker than I anticipated 😀
I'll need to have some process dicussions as well as find a reviewer, but this is a great start. I'm out of the office until next week but will follow up when I get back.

Just curious, how difficult was the translation process? I take it you were already familiar with Jupyter notebooks and TensorFlow? As we build out a community process, is this something you're interested in contributing to?

**Comment by 0101011** [comment]
Hi Billy,

1. I am not an X-Man, but quite familiar with TensorFlow and Jupyter and eager to contribute more:
**Is there a path I should follow (tutorial folder, what next?)**

2. Translation process is quite interesting tbh:

While writing, in my head I got the image of 16-20 years-old student reading this tutorial and tried to use the best suited language to make it understandable. 

- [ ] But sure do find a reviewer: I would appreciate some feedback! :)

**Comment by 0101011** [comment]
Four files added in total for review

**Comment by lamberta** [comment]
Thanks, Andrew! Working on a reviewer now.

Preview:

- https://github.com/0101011/docs/blob/master/site/ru/tutorials/keras/index.md
- https://colab.research.google.com/github/0101011/docs/blob/master/site/ru/tutorials/keras/basic_classification.ipynb
- https://colab.research.google.com/github/0101011/docs/blob/master/site/ru/tutorials/keras/basic_text_classification.ipynb

But I think [basic_regression.ipynb](https://github.com/0101011/docs/blob/master/site/ru/tutorials/keras/basic_regression.ipynb) may have a formatting error:
- https://colab.research.google.com/github/0101011/docs/blob/master/site/ru/tutorials/keras/basic_regression.ipynb
Can you take a look at that?

Thanks

**Comment by 0101011** [comment]
@lamberta, yup, it was a typo: the page is rendering on Colab now

**Comment by 0101011** [comment]
@tatianashp, Hi Tatiana,

I will commit more translations over the weekend as well as some possible amendments to the vocabulary used in the previous docs

File: site/ru/tutorials/keras/basic_text_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,838 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "35jv_fzP-llU"
+   },
+   "source": [
+    "## Обучаем модель\n",
+    "\n",
+    "Начнем тренировку нашей модели с 40 эпох при помощи мини-пакетов по 512 образцов. Это означает, что мы сделаем 40 итераций (проходов) по всем образцам данных в тензорах `x_train` и `y_train`. После обучения мы узнаем потери и точность нашей модели, показав ей 10,000 образцов из проверочного набора данных:"
```
[the remark this answers is not in the document]
Comment (0101011) [reply site/ru/tutorials/keras/basic_text_classification.ipynb]: Согласен, как раз насчёт них были сомнения, так как до переводов читал исключительно англоязычные доки

Спасибо за правку 👍

File: site/ru/tutorials/keras/basic_classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,1039 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "gut8A_7rCaW6"
+   },
+   "source": [
+    "Первый слой в нашей нейросети - `tf.keras.layers.Flatten` (flatten, пер. \"Выравнивать\"), будеn трансформировать изображения из 2D-массива (28 x 28 пикселей), в 1D-массив из 28 * 28 = 784 пикселей. Представь, что в этом слое мы как будто разбираем ряды сложенных пикселей изображения и выкладываем их в ряд. Этот слой не имеет никаких параметров для обучения; он просто преобразует формат данных.\n",
+    "\n",
+    "После того, как мы разложили все пиксели, нейросеть будет состоять из двух Dense (пер. \"Плотный\") `tf.keras.layers.Dense` слоев. Это так называемые полносвязные слои. Первый `Dense` слой содержит 128 нодов (или нейронов). Второй и последний *softmax* слой — возвращает массив с вероятностной шкалой которая должна равняться 1. Каждый нейрон содержит оценку, которая указывает вероятность принадлежности изображения к одному из 10 классов.\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_classification.ipynb]: `нодов` -> `узлов`? (graph node == узел графа)

File: site/ru/tutorials/keras/basic_classification.ipynb [hunk 3]
```diff
@@ -0,0 +1,1039 @@
...
+    "\n",
+    "### Компилируем модель\n",
+    "\n",
+    "Прежде чем модель будет готова для обучения, нам нужно указать еще несколько настроек. Их необходимо добавить в блоке *compile*:\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_classification.ipynb]: `указать еще несколько настроек` -> `несколько параметров`?

File: site/ru/tutorials/keras/basic_classification.ipynb [hunk 4]
```diff
@@ -0,0 +1,1039 @@
...
+    "\n",
+    "* *Loss function* (пер. \"Функция потерь\") — измеряет точность модели во время обучения. Нам важно чтобы *loss function* была как можно ниже, чтобы \"направить\" модель в нужном направлении\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_classification.ipynb]: `Нам важно чтобы *loss function* была как можно ниже, чтобы \"направить\" модель в нужном направлении`

->

`Мы хотим минимизировать значение этой фукнции, чтобы \"направить\" модель в нужном направлении`

File: site/ru/tutorials/keras/basic_classification.ipynb [hunk 5]
```diff
@@ -0,0 +1,1039 @@
...
+   "metadata": {
+    "colab_type": "text",
+    "id": "qKF6uW-BCaW-"
+   },
+   "source": [
+    "## Обучение модели\n",
+    "\n",
+    "Обучение модели нейронной сети требует выполнения следующих шагов:\n",
+    "\n",
+    "1. \"Накормить\" модель нашими данными для тренировки, в нашем случае - массивами `train_images` и `train_labels`\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_classification.ipynb]: `Накормить` совсем как то режет глаз. Может быть что то вроде "Загрузить тренировочные данные в модель, в нашем..."

File: site/ru/tutorials/keras/basic_classification.ipynb [hunk 6]
```diff
@@ -0,0 +1,1039 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "kgdvGD52CaXR"
+   },
+   "source": [
+    "Теперь давай отобразим побольше изображений и их предсказанные классы:\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_classification.ipynb]: `давай отобразим побольше изображений` -> `давай отобразим несколько изображений`

File: site/ru/tutorials/keras/basic_regression.ipynb [hunk 7]
```diff
@@ -0,0 +1,595 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "F_72b0LCNbjx"
+   },
+   "source": [
+    "## Загружаем данные о ценах на жилье Бостоне\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_regression.ipynb]: "с ценами на"

File: site/ru/tutorials/keras/basic_text_classification.ipynb [hunk 8]
```diff
@@ -0,0 +1,838 @@
...
+   "metadata": {
+    "colab_type": "text",
+    "id": "lFP_XKVRp4_S"
+   },
+   "source": [
+    "## Подготавливаем данные\n",
+    "\n",
+    "Обзоры фильмов из массива целых чисел должны быть конвертированы в тензоры прежде, чем они будут пропущены через нейросеть. Эта конвертация может быть сделана несколькими способами:\n",
+    "\n",
+    "* Прямое кодирование *one-hot encoding* конвертирует массивы в векторы 0 и 1. Например, последовательность [3, 5] станет 10,000-мерным вектором, полностью состоящим из нулей кроме показателей 3 и 5, которые будут представлены единицами. Затем, нам нужно будет создать первый `Dense` слой в нашей сети, который сможет принимать векторые данные с плавающей запятой. Такой подход очень требователен к объему памяти, несмотря на то, что требует указать размеры матрицы `num_words * num_reviews`\n",
```
Comment (ezhulenev) [inline site/ru/tutorials/keras/basic_text_classification.ipynb]: Судя по статьям на habr.com "One-Hot encoding" никто особо не переводит ("one hot encoding site:habr.com"), мне не кажется "прямое кодирование" облегчает понимание. Но я и статей на русском по этой теме никогда не читал, по этому не знаю как принято.

File: site/ru/tutorials/keras/basic_classification.ipynb (hunk 5 above)
Comment (0101011) [reply site/ru/tutorials/keras/basic_classification.ipynb]: Согласен :)

File: site/ru/tutorials/keras/basic_text_classification.ipynb (hunk 8 above)
Comment (0101011) [reply site/ru/tutorials/keras/basic_text_classification.ipynb]: Я читал, и там все плохо: 

Таким терминам будет лучше просто давать толкование, а далее в тексте использовать оригинал (тот же "one-hot encoding") - так будет проще понимать о каком методе идет речь

Передаем привет случайному лесу и методу ближайших соседей :)

Comment (ezhulenev) [reply site/ru/tutorials/keras/basic_text_classification.ipynb]: Метод ближайших соседей кстати еще вполне нормально. Я думаю что можно пользоваться вики чтобы проверять принято или нет переводить термин: https://ru.wikipedia.org/wiki/Random_forest, https://ru.wikipedia.org/wiki/%D0%9C%D0%B5%D1%82%D0%BE%D0%B4_k-%D0%B1%D0%BB%D0%B8%D0%B6%D0%B0%D0%B9%D1%88%D0%B8%D1%85_%D1%81%D0%BE%D1%81%D0%B5%D0%B4%D0%B5%D0%B9. Хотя конечно с One-Hot дичь какая то: https://ru.wikipedia.org/wiki/%D0%A3%D0%BD%D0%B8%D1%82%D0%B0%D1%80%D0%BD%D1%8B%D0%B9_%D0%BA%D0%BE%D0%B4. Пускай лучше будет прямое кодирование.

**Comment by ezhulenev** [comment]
@lamberta can we share a "style guide" we used for other translations? @0101011 we basically have a small guide with "common terminology" that should not be translated, my guess that 90% of it will be directly applicable to russian translation

File: site/ru/tutorials/keras/basic_text_classification.ipynb (hunk 8 above)
Comment (0101011) [reply site/ru/tutorials/keras/basic_text_classification.ipynb]: Да, как раз на Вики очень грамотные тексты по нейронкам, но далеко не все: 

1. На Хабре и других тематических сайтах в основном не увлекаются и пишут как есть, получается практично

2. Есть еще офицальные переводы книг Шолле и Гудфеллоу, но там [можно сломать голову](https://books.google.ru/books?id=97ZaDwAAQBAJ&printsec=frontcover&vq=embedding&hl=ru) в некоторых главах

**Comment by lamberta** [comment]
Thanks for your work on this guys, it's been a pretty educational experience :)

After talking with Eugene, it does seem like there are technical words in English that do not have a natural translation, and we suspect many of these words are common across many target languages (based on feedback from some Chinese reviewers, as well). We don't have a master cheat sheet of "do not translate" words, but I just created a wiki page so maybe we can start adding to that: https://github.com/tensorflow/docs/wiki/Translation-notes

Andrew, for the cheat sheets, do you mean like a glossary of translations for ML terms? That could be interesting. The Machine Learning Crash Course maintains an [ML glossary](https://developers.google.com/machine-learning/glossary/), and even some translations, but it doesn't look like Russian is there.

**Comment by 0101011** [comment]
@lamberta

1. The wiki guide for translation is great: we'll add the most arguable terms over the course of the next week

2. Downside: I can't PR to wiki of TF/docs, so we will drop you a list here as it's done

3. We can implement cheat sheets into docs and colabs as little snippets or blocks (think of college textbooks) with short explanations of newly meet words (can be done with vanilla markdown):

My guess is it'd be quite nifty for Korean, Chinese, Russian and other major target languages (btw, I saw an Arabic ML community on Twitter, ages 19 - 27)

4. Alternative is a separate glossary as ipynb doc, but I'll come back with a more practical example later

5. [Here's a cool wiki page](https://en.wikipedia.org/wiki/List_of_dictionaries_by_number_of_words) to get a sense of the approximate number of words in languages and why sometimes it is a problem :)

**Comment by lamberta** [comment]
Hi Andrew,
We recently updated the [Keras regression tutorial](https://www.tensorflow.org/tutorials/keras/basic_regression). The format is mostly the same but the content has changed to make the problem a little more straightforward. Would you mind updating your translation to reflect the current tensorflow.org? This section should now be relatively stable.

Yes, feel free to add your list here and I can update that wiki page. I'm not sure if we're going to keep that, but it should do for now. I'd prefer we standardize around the [MLCC glossary](https://developers.google.com/machine-learning/glossary/) but it doesn't look like there's a Russian version :(

Hold off on the cheatsheets, it will be easier to maintain these translations if they follow the en/ versions.

Thanks!

**Comment by 0101011** [comment]
@lamberta agree, maintaining should be stable

Got some fixes coming in 10 hours, I’ll be back 👍

**Comment by 0101011** [comment]
@lamberta Hi Billy,

1. Sure, no rush: I’m planning to do some final proofreading tomorrow just in case

2. I’ll subscribe to the mailing list in a minute to see how we can make use of it (discuss and developers lists are buzzing, so I guess that’ll work :)

Have a great weekend!

**Comment by lamberta** [comment]
Sounds good. Feel free to create another pull request if you have more edits. Thanks!
